sdk/keyvault/Azure.Security.KeyVault.Secrets/README.md

```diff
@@ -167,58 +168,71 @@ Console.WriteLine(secret.Value);
 
 ### Delete and purge a secret
 You will need to wait for the long-running operation to complete before trying to purge or recover the secret.
+You can do this by calling `UpdateStatus` in a loop as shown below:
 
 ```C# Snippet:DeleteAndPurgeSecret
-DeleteSecretOperation operation = await client.StartDeleteSecretAsync("secret-name");
+DeleteSecretOperation operation = client.StartDeleteSecret("secret-name");
 
 // You only need to wait for completion if you want to purge or recover the secret.
-await operation.WaitForCompletionAsync();
+while (!operation.HasCompleted)
+{
+    Thread.Sleep(2000);
```

Comment:
I think this is the wrong thing to show to people. Even `WaitForCompletionAsync().Wait()` is better

Reply:
Then why didn't we add a `WaitForCompletion` (sync) helper method? It seems odd to mix sync and async code like this in a sample, especially when it was discussed offline that we should highlight sync code instead of async, which, IMO, leads people down the wrong path.

Reply:
Per discussion, I'll leave as-is for now but will add a comment explaining you might want to call `UpdateStatus` after doing some other work or in another thread. But this also underscores why we should lead people down the path of async, for which I updated azure-sdk/issues#769.